go/connector/container.go

```diff
@@ -34,6 +41,20 @@ type Container struct {
 	waitCh chan error
 }
 
+// ExposePorts is a handle to a network configuration process that can expose
+// the ports for a running container. This is currently done by `runtime.ProxyServer`,
+// which proxies traffic throught the `grpc.ClientConn` to the container.
+type ExposePorts interface {
```

Comment:
Is this interface only to resolve an import cycle from `runtime`?

Reply:
Yeah, pretty much. I'd also considered making it into a more generic `ContainerObserver` interface, but I resisted the urge to prematurely generalize it.

Reply:
Gotcha. Ideally we could use the type directly instead of the interface IMO but not worth the refactoring to resolve the import cycle.